hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -465,7 +463,7 @@ private SnapshotDiffReportOzone createPageResponse(
     String tokenString = hasMoreEntries ? String.valueOf(idx) : null;
 
     if (!hasMoreEntries) {
-      checkReportsIntegrity(snapDiffJob, idx);
+      checkReportsIntegrity(snapDiffJob, index, diffReportList.size());
```

Comment:
For now, I'm OK to with the check. Because it will eventually fail for the below scenario but it should fail fast.

Case: 
SnapshotDiffJob.TotalEntries = 1000, PageSize = 1000, Report iterators says 2500+ (<3000)  entries (due to some corruption (or some bug). 

At the end of the first page response, `checkReportsIntegrity` check will be skipped and client will call again for second page. `checkReportsIntegrity`  check will be skipped again at the end of second page response and client will call for third page. At the end of the third page request will fail because eventually it will call `checkReportsIntegrity`.